src/herder/HerderImpl.cpp

```diff
@@ -2192,15 +2180,64 @@ HerderImpl::trackingHeartBeat()
 }
 
 void
-HerderImpl::updateTransactionQueue(TxSetFrameConstPtr txSet)
+HerderImpl::updateTransactionQueue(TxSetFrameConstPtr externalizedTxSet)
 {
     ZoneScoped;
+    if (externalizedTxSet == nullptr)
+    {
+        CLOG_DEBUG(Herder,
+                   "No tx set to update tx queue - expected during bootstrap");
+        return;
+    }
+    TxSetFrame::Transactions const* classicTxs = nullptr;
+    TxSetFrame::Transactions const* sorobanTxs = nullptr;
+    TxSetFrame::Transactions txsForNonActiveTxSet;
+
+    // Try to extract the transactions from the tx set.
+    // These transactions are used for tx queue cleanup.
+    if (externalizedTxSet->getResolvedFrame() != nullptr ||
```

Comment:
Because conceptually relying on a resolved tx set when it's not needed is confusing. Also this code now looks a lot more complex, handling two different conditions that can be handled with one. This happens once a ledger, how much are we saving really?

Reply:
Ok, changed this to only use XDR tx sets.